chia/plotters/bladebit.py

```diff
@@ -0,0 +1,246 @@
+    supported: bool = is_bladebit_supported()
+
+    if get_bladebit_executable_path(plotters_root_path).exists():
+        version: Optional[str] = None
+        try:
+            proc = subprocess.run(
+                [os.fspath(get_bladebit_executable_path(plotters_root_path)), "--version"],
+                capture_output=True,
+                text=True,
+            )
+            version = proc.stdout.strip()
+        except Exception as e:
+            print(f"Failed to determine bladebit version: {e}")
+
+        if version is not None:
+            installed = True
+            info["version"] = version
+        else:
+            installed = False
+
+    info["installed"] = installed
+    if installed is False:
+        info["can_install"] = supported
+
+    if supported and meets_memory_requirement() is False:
+        info["bladebit_memory_warning"] = f"BladeBit requires at least {BLADEBIT_MIN_RAM_GIBS} GiB of RAM to operate"
+
+    return info
+
+
+progress = {
+    "Finished F1 sort": 0.01,
+    "Finished forward propagating table 2": 0.06,
+    "Finished forward propagating table 3": 0.12,
+    "Finished forward propagating table 4": 0.2,
+    "Finished forward propagating table 5": 0.28,
+    "Finished forward propagating table 6": 0.36,
+    "Finished forward propagating table 7": 0.42,
+    "Finished prunning table 6": 0.43,
+    "Finished prunning table 5": 0.48,
+    "Finished prunning table 4": 0.51,
+    "Finished prunning table 3": 0.55,
+    "Finished prunning table 2": 0.58,
+    "Finished compressing tables 1 and 2": 0.66,
+    "Finished compressing tables 2 and 3": 0.73,
+    "Finished compressing tables 3 and 4": 0.79,
+    "Finished compressing tables 4 and 5": 0.85,
+    "Finished compressing tables 5 and 6": 0.92,
+    "Finished compressing tables 6 and 7": 0.98,
+}
+
+
+# https://kevinmccarthy.org/2016/07/25/streaming-subprocess-stdin-and-stdout-with-asyncio-in-python/
+async def _read_stream(stream, callback):
+    while True:
+        line = await stream.readline()
+        if line:
+            callback(line)
+        else:
+            break
+
+
+def parse_stdout(out):
+    out = out.rstrip()
+    print(out)
+    for k, v in progress.items():
+        if k in out:
+            print(f"Progress update: {v}")
+
+
+async def run(command):
+    process = await asyncio.create_subprocess_shell(
+        command, stdout=asyncio.subprocess.PIPE, stderr=asyncio.subprocess.PIPE
+    )
+
+    await asyncio.wait(
+        [
+            _read_stream(
+                process.stdout,
+                lambda x: parse_stdout(x.decode("UTF8")),
+            ),
+            _read_stream(
+                process.stderr,
+                lambda x: print("STDERR: {}".format(x.decode("UTF8"))),
+            ),
+        ]
+    )
+
+    await process.wait()
+
+
+async def run_bladebit(args):
+    cmd = ""
+    for arg in args:
+        cmd += arg
+        cmd += " "
+    print(f"Running command: {cmd[:-1]}")
```

Comment:
Why is the last arg left out?

Reply:
it's a space character.

Reply:
minor: could use `" ".join(args)` which would avoid the trailing space.

Reply:
Good point @altendky re: escaping and params with spaces. Would subprocess.run() be a safer option?

Reply:
As noted in https://github.com/Chia-Network/chia-blockchain/pull/8497/files/4daee91fa3e296745cc09ee6bb481e013ee3dfd9#r713366777, we probably shouldn't be using shells.  And yeah, part of that is not joining the arguments into a single string but instead keeping them in a list.  For this print, it's just a matter of printing the output confusingly.  For the actual execution, yes, we should use the existing `args` list without having joined it.

Note though that `subprocess.run()` doesn't clarify between using a shell or not.  That's the `shell=` option to `.run()`.  Also, this is async and so needs to use an sync call, not the `subprocess` module directly.

Reply:
@altendky I believe this has been addressed by @fchirica's refactoring. I tried plotting with some temp/final paths that included spaces in the name. Worked fine.

Reply:
Agreed.  I should be keeping on top of the changes and resolving my discussions.  Thanks for getting me back here.